File: node-app/tictactoe/cell.js
function cell(){
    var obj={};

    obj.cellValue="";

	obj.getValue=function() {
		if(obj.cellValue==""){
			return "";
		}
		return obj.cellValue;
	}

	obj.setValue=function(mark) {
		if(obj.cellValue=="")
		    obj.cellValue=mark;
		else
			console.log("Cell already marked");
			
	}

    return obj;
}
module.exports=cell;

File: TicTacToe/app/src/main/java/com/techlabs/paresh/tic_tac_toe/Cell.java
package com.techlabs.paresh.tic_tac_toe;

public class Cell {
	public Mark cellValue=Mark.EMPTY;

	public Mark getValue() {
		if(this.cellValue==null){
			return Mark.EMPTY;
		}
		return cellValue;
	}

	public void setValue(Mark mark) {
			try{
			if(cellValue.equals(Mark.EMPTY))
				cellValue=mark;
			else
				throw new Exception();
			}
			catch(Exception e){
				System.out.println("Cell already marked");
			}
	}
	
	
	
}
